windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -295,20 +288,59 @@ func (e executor) shutdownHost(ctx context.Context, cmd *landscapeapi.Command_Sh
 	return gowsl.Shutdown(ctx)
 }
 
-func installFromMicrosoftStore(ctx context.Context, distro gowsl.Distro) (err error) {
+// installFromWSLOnlineDistros installs a distro by means of `wsl --install`, which can be either an appx from MS Store or a modern distro from anywhere else.
+func installFromWSLOnlineDistros(ctx context.Context, distro gowsl.Distro) (err error) {
 	defer decorate.OnError(&err, "can't install from Microsoft Store")
 
 	if err := gowsl.Install(ctx, distro.Name()); err != nil {
 		return err
 	}
 
+	// Prioritize the old distro installation method: via the appx distro launcher.
 	if err := distroinstall.InstallFromExecutable(ctx, distro); err != nil {
+		var cmdErr *distroinstall.CommandNotFoundError
+		if errors.As(err, &cmdErr) {
+			// If gowsl.Install succeeded but the distro launcher is not found, this is likely a "modern distro".
+			log.Debugf(ctx, "Distro launcher not found, trying as tar-based distro: %s", cmdErr)
+			return installModernDistro(ctx, distro)
+		}
 		return err
 	}
 
 	return nil
 }
 
+// installModernDistro finishes setting up a modern distro by waiting for cloud-init to finish.
+func installModernDistro(ctx context.Context, distro gowsl.Distro) (err error) {
+	defer decorate.OnError(&err, "can't install modern distro")
+
+	err = touchdistro.WaitForCloudInit(ctx, distro.Name())
+
+	return err
+}
+
+// createDefaultUser creates a default user whose name is derived from the current Windows user.
+func createDefaultUser(ctx context.Context, distro gowsl.Distro) (err error) {
+	userName := "ubuntu"
+
+	windowsUser, err := user.Current()
+	if err != nil {
+		return err
+	}
+	userFields := strings.Split(windowsUser.Username, "\\") // Typically "hostname\\username"
+	n := userFields[len(userFields)-1]
```

Comment:
```suggestion
	n := userFields[len(userFields) - 1]
```

Reply:
The go formatter doesn't like your suggestion.

Reply:
Weird since it seems like we use spaces between operands elsewhere